pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/model/XPathSuggestions.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.util.fxdesigner.model;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.List;
+
+
+
+
+public class XPathSuggestions {
+    private List<String> xPathSuggestions = new ArrayList<>();
+    private String packagename;
+    private String language;
+
+    public XPathSuggestions(String language) {
+        this.packagename = "net.sourceforge.pmd.lang." + language + ".ast";
+        this.language = language;
+    }
+
+
+    public List<String> getXPathSuggestions() throws IOException, ClassNotFoundException {
+        return evaluateXpathSuggestions(createList(getClasses(packagename)));
```

Comment:
So you reevaluate everything each time. Keep in mind your list of all node names only be evaluated once, eg in the constructor. Also, throwing checked exceptions here seems really sloppy. The exceptions should be taken care of when the list is created in the constructor, and should be recovered from in this class. Clients of this class have no idea how to recover from them